'Update Event' Google Calendar connector action input shape issue

## This is a...

<pre><code>
[ ] Feature request
[ ] Regression (a behavior that used to work and stopped working in a new release)
[x] Bug report  <!-- Please search GitHub for a similar issue or PR before submitting -->
[ ] Documentation issue or request
</code></pre>

## Description
I have troubles using the Update Event (io.syndesis:google-calendar-update-event-connector) action as an end step.

The action itself defines properties, e.g. eventId, which needs to be specified for Event identification. Its input shape is com.google.api.services.calendar.model.Event. When the preceding step is a mapper step, then I can map to all the fields of com.google.api.services.calendar.model.Event, but I can't fill in any of the properties (eventId in my case).

When I map only to Event class fields (i.e. not eventId property) I get following error:
```
org.apache.camel.RuntimeCamelException: Error invoking update with {eventId=null, calendarId=jbossqa.fuse@gmail.com, content={id=1pig4rhkl3qtji5s9l8e2l9f3j, kind=calendar#event, location=new-location, status=confirmed, summary=test1}}: Required parameter eventId must be specified.
--
at org.apache.camel.util.component.ApiMethodHelper.invokeMethod(ApiMethodHelper.java:514) ~[camel-core-2.21.0.fuse-720021.jar!/:2.21.0.fuse-720021]
at org.apache.camel.util.component.AbstractApiProducer.doInvokeMethod(AbstractApiProducer.java:120) ~[camel-core-2.21.0.fuse-720021.jar!/:2.21.0.fuse-720021]
at org.apache.camel.component.google.calendar.GoogleCalendarProducer.doInvokeMethod(GoogleCalendarProducer.java:43) ~[camel-google-calendar-2.21.0.fuse-720021.jar!/:2.21.0.fuse-720021]
at org.apache.camel.util.component.AbstractApiProducer$1.run(AbstractApiProducer.java:86) ~[camel-core-2.21.0.fuse-720021.jar!/:2.21.0.fuse-720021]
at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511) [na:1.8.0_151]
at java.util.concurrent.FutureTask.run(FutureTask.java:266) [na:1.8.0_151]
at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.access$201(ScheduledThreadPoolExecutor.java:180) [na:1.8.0_151]
at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:293) [na:1.8.0_151]
at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149) [na:1.8.0_151]
at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624) [na:1.8.0_151]
at java.lang.Thread.run(Thread.java:748) [na:1.8.0_151]
Caused by: java.lang.NullPointerException: Required parameter eventId must be specified.
at com.google.api.client.repackaged.com.google.common.base.Preconditions.checkNotNull(Preconditions.java:229) ~[google-http-client-1.22.0.jar!/:1.22.0]
at com.google.api.client.util.Preconditions.checkNotNull(Preconditions.java:140) ~[google-http-client-1.22.0.jar!/:1.22.0]
at com.google.api.services.calendar.Calendar$Events$Update.<init>(Calendar.java:5492) ~[google-api-services-calendar-v3-rev291-1.22.0.jar!/:na]
at com.google.api.services.calendar.Calendar$Events.update(Calendar.java:5463) ~[google-api-services-calendar-v3-rev291-1.22.0.jar!/:na]
at sun.reflect.GeneratedMethodAccessor407.invoke(Unknown Source) ~[na:na]
at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[na:1.8.0_151]
at java.lang.reflect.Method.invoke(Method.java:498) ~[na:1.8.0_151]
at org.apache.camel.util.component.ApiMethodHelper.invokeMethod(ApiMethodHelper.java:506) ~[camel-core-2.21.0.fuse-720021.jar!/:2.21.0.fuse-720021]
... 10 common frames omitted
```
I believe this is a valid use case, when the flow starts with google calendar 'From' action and ending with google calendar 'To' action, for example when user wants to change locations of all events scheduled to another location (due to its unavailability or sth).

--- a/app/connector/google-calendar/src/main/java/io/syndesis/connector/calendar/GoogleCalendarUpdateEventCustomizer.java
+++ b/app/connector/google-calendar/src/main/java/io/syndesis/connector/calendar/GoogleCalendarUpdateEventCustomizer.java
@@ -77,6 +77,7 @@ public class GoogleCalendarUpdateEventCustomizer implements ComponentProxyCustom
         options.put("methodName", "update");
     }
 
+    @SuppressWarnings("PMD.NPathComplexity")
     private void beforeProducer(Exchange exchange) throws MessagingException, IOException, ParseException {
 
         final Message in = exchange.getIn();

--- a/app/connector/google-calendar/src/main/java/io/syndesis/connector/calendar/GoogleCalendarUpdateEventCustomizer.java
+++ b/app/connector/google-calendar/src/main/java/io/syndesis/connector/calendar/GoogleCalendarUpdateEventCustomizer.java
@@ -110,6 +110,9 @@ public class GoogleCalendarUpdateEventCustomizer implements ComponentProxyCustom
             if (ObjectHelper.isNotEmpty(location)) {
                 event.setLocation(location);
             }
+            if (ObjectHelper.isEmpty(eventId)) {
+                eventId = event.getId();
+            }
         }
 
         in.setHeader("CamelGoogleCalendar.content", event);

